Lib/typing.py

```diff
@@ -835,7 +836,11 @@ def __new__(cls, *args, **kwds):
         if cls is Generic:
             raise TypeError("Type Generic cannot be instantiated; "
                             "it can be used only as a base class")
-        return super().__new__(cls)
+        if super().__new__ is object.__new__:
```

Comment:
I think ` and cls.__init__ is not object.__init__` is needed here.

The test:
```python
        class A(Generic[T]):
            pass
        with self.assertRaises(TypeError):
            A('foo')
```
And would extract the subexpression `super().__new__` into a variable.

Reply:
Hm, indeed, in the original PR in typing  repo we were concerned that it may fail when it shouldn't, but forgot that it also should fail when it should. @gvanrossum what do you think?

Reply:
Yes, @serhiy-storchaka is right. This is definitely a wart -- `object.__init__` and `object.__new__` each ignore extra arguments only when the other is overridden (and they themselves are not).